modules/launcher/src/main/java/org/mule/module/launcher/util/ObservableList.java

```diff
@@ -323,14 +324,16 @@ public boolean hasNext()
         @Override
         public E next()
         {
-            cursor++;
-            return iterDelegate.next();
+            E next = iterDelegate.next();
+            lastRet = cursor++;
+            return next;
         }
 
         @Override
         public void remove()
         {
             ObservableList.this.remove(cursor--);
+	    lastRet = -1;
```

Comment:
please replace all occurences of -1 with a constant. That constant should be private and final